COMETwebapp/Components/Viewer/BabylonCanvasBase.cs

```diff
@@ -0,0 +1,169 @@
+                cube.SetRotation(1.0, 0.2, 0.1);
+                Scene.AddPrimitive(cube, Color.Red);
+
+                CustomPrimitive cp = new CustomPrimitive("./Assets/obj/", "RX2_CUSTOM_BODYKIT.obj");
+                Scene.AddPrimitive(cp);
+            }
+        }
+
+        /// <summary>
+        /// Initialize the elements in the scene based on the elements of the iteration
+        /// </summary>
+        private void InitializeElements()
+        {
+            var iteration = this.SessionAnchor.OpenIteration;
+            var elementUsages = iteration?.Element.SelectMany(x => x.ContainedElement).ToList();
+
+            if(elementUsages != null)
+            {
+                foreach (var elementUsage in elementUsages)
+                {
+                    CreateShapeBasedOnElementUsage(elementUsage);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates a shape for the scene based on the element usage
+        /// </summary>
+        /// <param name="elementUsage">The element usage used for creating the shape</param>
+        private void CreateShapeBasedOnElementUsage(ElementUsage elementUsage)
+        {
+            //elementUsage.ParameterOverride.Find(x=>x.p)
+            if (elementUsage.ParameterOverride.Count > 0)
+            {
+
+            }
+        }
+
+        /// <summary>
+        /// Canvas on mouse down event
+        /// </summary>
+        /// <param name="e">the mouse args of the event</param>
+        public void OnMouseDown(MouseEventArgs e)
+        {
+
+        }
+
+        /// <summary>
+        /// Canvas on mouse up event
+        /// </summary>
+        /// <param name="e">the mouse args of the event</param>
+        public void OnMouseUp(MouseEventArgs e)
+        {
+
+        }
+
+        /// <summary>
+        /// Canvas on click event
+        /// </summary>
+        /// <param name="e">the mouse args of the event</param>
+        public void OnClick(MouseEventArgs e)
+        {
+
+        }
+
+        /// <summary>
+        /// Canvas on mouse move event
+        /// </summary>
+        /// <param name="e">the mouse args of the event</param>
+        public void OnMouseMove(MouseEventArgs e) 
+        { 
+
+        }
+
+        /// <summary>
+        /// Canvas on mouse wheel event
+        /// </summary>
+        /// <param name="e">the mouse wheel args of the event</param>
+        public void OnMouseWheel(WheelEventArgs e)
+        {
+            Scene.SetInfoPanelVisibility(false);
+        }
+
+        /// <summary>
+        /// Canvas on key down event
+        /// </summary>
+        /// <param name="e">the keyboard args of the event</param>
+        public void OnKeyDown(KeyboardEventArgs e)
+        {
+
+        }
+
+        /// <summary>
+        /// Canvas on key up event
+        /// </summary>
+        /// <param name="e">the keyboard args of the event</param>
+        public void OnKeyUp(KeyboardEventArgs e)
+        {
+
```

Comment:
I propose you put a `//TODO:` statement here to make it explicit what you intend to code here.